espnet2/gan_tts/hifigan/hifigan.py

```diff
@@ -572,6 +577,86 @@ def _apply_spectral_norm(m: torch.nn.Module):
 
         self.apply(_apply_spectral_norm)
 
+    def remove_weight_norm(self):
+        """Remove weight normalization module from all of the layers."""
+
+        def _remove_weight_norm(m):
+            try:
+                logging.debug(f"Weight norm is removed from {m}.")
+                torch.nn.utils.remove_weight_norm(m)
+            except ValueError:  # this module didn't have weight norm
+                return
+
+        self.apply(_remove_weight_norm)
+
+    def remove_spectral_norm(self):
+        """Remove spectral normalization module from all of the layers."""
+
+        def _remove_spectral_norm(m):
+            try:
+                logging.debug(f"Spectral norm is removed from {m}.")
+                torch.nn.utils.remove_spectral_norm(m)
+            except ValueError:  # this module didn't have weight norm
+                return
+
+        self.apply(_remove_spectral_norm)
+
+    def _load_state_dict_pre_hook(
+        self,
+        state_dict,
```

Comment:
@iamanigeeit I think the `state_dict` here includes only keys corresponding to the current modules:
```python
(Pdb++) state_dict.keys()
dict_keys(['msd.discriminators.0.layers.0.0.weight', 'msd.discriminators.0.layers.0.0.bias', 'msd.discriminators.0.layers.1.0.weight', 'msd.discriminators.0.layers.1.0.bias', 'msd.discriminators.0.layers.2.0.weight', 'msd.discriminators.0.layers.2.0.bias', 'msd.discriminators.0.layers.3.0.weight', 'msd.discriminators.0.layers.3.0.bias', 'msd.discriminators.0.layers.4.0.weight', 'msd.discriminators.0.layers.4.0.bias', 'msd.discriminators.0.layers.5.0.weight', 'msd.discriminators.0.layers.5.0.bias', 'msd.discriminators.0.layers.6.0.weight', 'msd.discriminators.0.layers.6.0.bias', 'msd.discriminators.0.layers.7.weight', 'msd.discriminators.0.layers.7.bias'])
```
So I think we do not need to filter. Could you check the current code?

Reply:
I added in `hifigan.py`
```
        for x in state_dict.keys():
            print(x)
        if self.use_weight_norm and not any(
            ["weight_g" in k for k in state_dict.keys()]
        ):
```

Then:
```
import os
os.environ['CUDA_LAUNCH_BLOCKING'] = "1"
from pathlib import Path
from espnet2.bin.tts_inference import Text2Speech
PWD = %pwd
PWD = Path(PWD)
LJSPEECH_DIR = (PWD / '../egs2/ljspeech/tts1/').resolve()
device = 'cuda'
os.chdir(LJSPEECH_DIR)
pretrained_dir = LJSPEECH_DIR / "exp/tts_train_jets_raw_phn_tacotron_g2p_en_no_space"
pretrained_model_file = pretrained_dir / "train.total_count.ave_5best.pth"
pretrained_tts = Text2Speech.from_pretrained(
    train_config=pretrained_dir / "config.yaml",
    model_file=pretrained_model_file,
    device=device
)
pretrained_model = pretrained_tts.model
os.chdir(PWD)
```

feats_extract.logmel.melmat
normalize.mean
normalize.std
pitch_normalize.mean
pitch_normalize.std
energy_normalize.mean
energy_normalize.std
tts.generator.encoder.embed.0.weight
tts.generator.encoder.embed.1.alpha
tts.generator.encoder.encoders.0.self_attn.linear_q.weight
tts.generator.encoder.encoders.0.self_attn.linear_q.bias
...
tts.discriminator.mpd.discriminators.4.output_conv.bias
tts.discriminator.mpd.discriminators.4.output_conv.weight_g
tts.discriminator.mpd.discriminators.4.output_conv.weight_v
tts.mel_loss.wav_to_mel.logmel.melmat

Reply:
Does it work for you ?
```python
        current_module_keys = [
            x for x in self.state_dict().keys() if x.startswith(prefix)
        ]
        if self.use_weight_norm and not any(
            ["weight_g" in k for k in current_module_keys]
        ):
```

Reply:
Sorry, i got disconnected due to network upgrade.

Yes, this works -- i am on PyTorch 1.11. Thanks!